integrations/vercel-ai-sdk.mdx

```diff
@@ -50,20 +53,12 @@ export async function streamAnswer(question: string) {
 
 ## With Threads and Steps
 
-You can pass a [Threads](/concepts/observability/thread) or a [Steps](/concepts/observability/step) as an option of your generations to keep your logs organized.
-
-<CodeGroup>
-```typescript TypeScript
-import { LiteralClient } from '@literalai/client';
-
-import { generateText as baseGenerateText } from 'ai';
-import { openai } from '@ai-sdk/openai';
-
-const literalClient = new LiteralClient(process.env['LITERAL_API_KEY']); // This is the default and can be omitted
+In most cases, you will want to keep track of the different generations from your application by grouping them into [Threads](/concepts/observability/thread) or [Steps](/concepts/observability/step). This is especially useful when you want to understand the context in which a generation was made, or when you want to compare different generations.
 
-const generateText = literalClient.instrumentation.vercel.instrument(baseGenerateText);
+You can attach generation to a `Thread` or a `Step` by passing it as the `literalAiParent` parameter. This parameter should be an instance of a `Thread` or a `Step` that you have created using the Literal AI SDK.
```

Comment:
I didn't know this `literalAiParent`, it feels like an additional layer compared to the Python client where we keep track of the current active thread/step. Is it a limitation of TypeScript or a limitation of Vercel AI SDK?

Reply:
I think it is linked to the fact that for the moment the Typescript SDK does not handle any sort of context. This should be addressed in [this ticket](https://linear.app/literalai/issue/ENG-1578/investigate-ts-decoratorscontext-manager-for-sdk) which is on my table this week